File: base/lib_base/src/main/java/com/pds/base/act/BaseActivity.java
package com.pds.base.act;

import android.os.Bundle;
import android.view.MotionEvent;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.pds.ui.dialog.LoadingDialog;
import com.pds.util.ui.ImmersiveModeUtil;

/**
 * @author: pengDaoSong
 * CreateTime:  2020-09-16 15:21
 * Description: 自定义顶层Activity基类，尽量不要包含特定业务处理
 */
public class BaseActivity extends AppCompatActivity {

    protected LoadingDialog mLoadingDialog;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        configImmersiveMode();
    }

    @Override
    protected void onResume() {
        super.onResume();
    }

    @Override
    public void setContentView(int layoutResID) {
        this.setContentView(getLayoutInflater().inflate(layoutResID, null));
    }

    public void showDialogLoading() {
        showDialogLoading(true);
    }

    public void showDialogLoading(boolean cancelable) {
        if (checkDialog(cancelable)) {
            return;
        }
        mLoadingDialog.show(getSupportFragmentManager());
    }

    public void showDialogLoadingDelay() {
        showDialogLoadingDelay(true);
    }

    public void showDialogLoadingDelay(boolean cancelable) {
        if (checkDialog(cancelable)) {
            return;
        }
        mLoadingDialog.showDelay(getSupportFragmentManager());
    }

    private boolean checkDialog(boolean cancelable) {
        if (null == mLoadingDialog) {
            mLoadingDialog = LoadingDialog.getInstance(cancelable);
        }
        if (isDestroyed() || mLoadingDialog.isAdded() || mLoadingDialog.isVisible()) {
            return true;
        }
        return false;
    }

    public void hideDialogLoading() {
        if (mLoadingDialog != null) {
            mLoadingDialog.dismissAllowingStateLoss();
            mLoadingDialog = null;
        }
    }

    /**
     * 配置沉浸式模式
     */
    protected void configImmersiveMode() {
        ImmersiveModeUtil.setDefaultImmersiveMode(this);
    }

    @Override
    protected void onStop() {
        super.onStop();
    }

    @Override
    public boolean dispatchTouchEvent(MotionEvent event) {
        return super.dispatchTouchEvent(event);
    }
}

File: business/phonequery/src/main/java/com/pds/phone/query/BaseActivity.kt
package com.pds.phone.query

import android.annotation.SuppressLint
import android.app.Activity
import android.content.ComponentName
import android.content.Intent
import android.content.pm.ApplicationInfo
import android.os.Bundle
import android.view.*
import androidx.annotation.NonNull
import com.pds.splugin.model.InterfaceActivity

/**
 * @author: pengdaosong
 * @CreateTime:  2020/11/15 2:11 PM
 * @Email: pengdaosong@medlinker.com
 * @Description:
 */
@SuppressLint("MissingSuperCall")
open class BaseActivity : Activity(), InterfaceActivity {

    protected var mThat: Activity? = null

    override fun attach(proxyActivity: Activity) {
        mThat = proxyActivity
    }

    override fun setContentView(view: View?) {
        if (mThat != null) {
            mThat?.setContentView(view)
        } else {
            super.setContentView(view)
        }
    }

    override fun setContentView(layoutResID: Int) {
        if (mThat != null) {
            mThat?.setContentView(layoutResID)
        } else {
            super.setContentView(layoutResID)
        }
    }

    override fun startService(service: Intent): ComponentName? {
        val m = Intent()
        m.putExtra("serviceName", service.component.className)
        return mThat?.startService(m)
    }

    override fun <T : View> findViewById(id: Int): T? {
        return if (null != mThat) mThat?.findViewById(id) else super.findViewById(id)
    }

    override fun getIntent(): Intent? {
        return if (null != mThat) mThat?.intent else super.getIntent()

    }

    override fun getClassLoader(): ClassLoader?{
        return if (null != mThat) mThat?.classLoader else super.getClassLoader()
    }

    override fun startActivity(intent: Intent) {
        val m = Intent()
        m.putExtra("className", intent.component.className)
        mThat?.startActivity(m)
    }

    @NonNull
    override fun getLayoutInflater(): LayoutInflater? {
        return if (null != mThat) mThat?.layoutInflater else super.getLayoutInflater()
    }

    override fun getApplicationInfo(): ApplicationInfo? {
        return if (null != mThat) mThat?.applicationInfo else super.getApplicationInfo()
    }
    override fun getWindow(): Window? {
        return if (null != mThat) mThat?.window else super.getWindow()
    }

    override fun getWindowManager(): WindowManager?{
        return if (null != mThat) mThat?.windowManager else super.getWindowManager()
    }


    override fun onCreate(savedInstanceState: Bundle) {}

    override fun onResume() {
    }

    override fun onPause() {
    }

    override fun onSaveInstanceState(outState: Bundle) {
    }

    override fun onStop() {
    }

    override fun onBackPressed() {
    }

    override fun onStart() {
    }

    override fun onDestroy() {
    }

}